File: ElectronicPrescription/Models/Posology.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations.Schema;

namespace ElectronicPrescription.Models
{
    public class Posology
    {
        [ForeignKey("Prescription")]
        public int PosologyId { get; set; }

        public int Quantity { get; set; }

        public string Technique { get; set; }

        public string Interval { get; set; }

        public string Period { get; set; }

        public ICollection<PackageLeaflet> PackageLeaflet { get; set; }
    }
}

File: frontend-electronic-prescriptions/src/app/model/posology.ts
export class Posology {
    quantity: string;
    technique: string;
    interval: string;
    period: string;

    constructor(
        quantity: string,
        technique: string,
        interval: string,
        period: string
    ) {
        this.quantity = quantity;
        this.technique = technique;
        this.interval = interval;
        this.period = period;
    }
}